graviti/dataframe/series.py

```diff
@@ -20,48 +20,93 @@ class Series(Generic[_T]):
         schema: Data type to force. Only a single dtype is allowed. If None, will be
             inferred from `data`.
         name: The name to the Series.
-        copy: Copy input data. Only affects Series or 1d ndarray input.
-        client: The client for getting a remote data.
 
     Examples:
         Constructing Series from a list.
 
-        >>> d = [1, 2, 3]
+        >>> d = {"filename": "a.jpg", "attributes": {"color": "red", "pose": "frontal"}}
         >>> series = Series(data=d)
         >>> series
-        0   1
-        1   2
-        2   3
-        schema: int64
+        filename         a.jpg
+        attributes color red
+                   pose  frontal
+        schema: string
 
     """
 
+    _data: List[Any]
+    _indices_location: Dict[_T, int]
+    _indices: List[_T]
+
     def __init__(  # pylint: disable=too-many-arguments
         self,
-        data: Optional[Iterable[Any]] = None,
+        data: Optional[Dict[_T, Any]] = None,
         schema: Any = None,
         name: Optional[str] = None,
-        copy: Optional[bool] = None,
-        client: Any = None,
     ) -> None:
-        pass
+        if data is None:
+            data = {}
+
+        self._init(data, schema)
+        self._indices_location = {value: key for key, value in enumerate(data.keys())}
+        self.name = name
+
+    # @overload
+    # def __getitem__(self, key: slice) -> "Series":
+    #    ...
+
+    @overload
+    def __getitem__(self, key: Tuple[_T]) -> "Series[_T]":
+        ...
 
     @overload
     def __getitem__(self, key: _T) -> Any:
         ...
 
     @overload
-    def __getitem__(self, key: Union[Iterable[_T], slice]) -> "Series[_T]":
+    def __getitem__(self, key: Iterable[_T]) -> "Series[_T]":
         ...
 
-    def __getitem__(self, key: Union[_T, Iterable[_T], slice]) -> Any:
-        pass
+    def __getitem__(self, key: Union[_T, Iterable[_T]]) -> Any:
+        if isinstance(key, (str, int)):
+            return self._data[self._indices_location[key]]
+
+        new_data = {name: self._data[self._indices_location[name]] for name in key}
+        return Series(new_data)
 
     def __repr__(self) -> str:
         pass
 
     def __len__(self) -> int:
-        pass
+        return len(self._data)
```

Comment:
Will `return self._data.__len__()` better, because the function is `__len__`, call `__len__` in `__len__` looks more  symmetrical